packages/ui5-task-compileless/lib/compileLess.js

```diff
@@ -0,0 +1,100 @@
+                        string: output.css
+                    })
+                ).catch(error => log.error(error));
+        })
+    );
+}
+
+/**
+ * Custom task to compile less files in the app folder
+ *
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {module:@ui5/fs.AbstractReader} parameters.dependencies Reader or Collection to read dependency files
+ * @param {Object} parameters.options Options
+ * @param {string} parameters.options.projectName Project name
+ * @param {string} [parameters.options.configuration] Task configuration if given in ui5.yaml
+ * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
+ */
+module.exports = async function({workspace, dependencies, options}) {
+    const appFolderPath = options.configuration && options.configuration.appFolderPath || 'webapp';
+    const lessToCompile = options.configuration && options.configuration.lessToCompile|| [] ;
+    const isDebug = options.configuration && options.configuration.debug;
+
+
+
+    //create custom duplex collection where the webapp is in the "/" folder
+    //By default when building the workspace reader is in the vir dir "/resources/{namespace}
+    //This needs to be done so that the less in the serve version and the build version can use the same paths to external resources
+    let appFolderWorkspace = new DuplexCollection({
+        reader: new FileSystem({
+            virBasePath: "/",
+            fsBasePath: appFolderPath
+        }),
+        writer: new Memory({
+            virBasePath: "/"
+        })
+    });
+
+    const customCombo = new ReaderCollectionPrioritized({
+        name: `${options.projectName} - prioritize app folder over dependencies`,
+        readers: [appFolderWorkspace, dependencies]
+    });
+
+
+    const lessResourcesProjectWorkspace = await workspace.byGlobSource("**/*.less");
+    let lessResourcesAppFolderWorkspace = await appFolderWorkspace.byGlobSource("**/*.less");
+
+    // apply include patterns
+    lessResourcesAppFolderWorkspace = lessResourcesAppFolderWorkspace.filter(appFolderResource => {
+        const resourcePath = appFolderResource.getPath();
+        return lessToCompile.some(excludePattern => resourcePath.includes(excludePattern) || minimatch(resourcePath.slice(1), excludePattern))
+    });
+
+    const lessResources = lessResourcesAppFolderWorkspace.map(appFolderResource => {
+        const lessResourceProjectWorkspace = lessResourcesProjectWorkspace.find(projectResource => projectResource.getPath().includes(appFolderResource.getPath()));
+        return {
+            projectResourcePath: lessResourceProjectWorkspace.getPath(),
+            resource: appFolderResource
+        };
+    });
+
+    const compiledResources = await compileLess(lessResources, fsInterface(customCombo), isDebug);
```

Comment:
see comment on fn signature of `compileLess`

Reply:
done